File: DesignPatternsExercises/java/ChainOfResponsibilityDesignPattern/Solution/SupportChain.java
// public __________(4)__________ class __________(5)__________ {
public abstract class SupportChain {
	protected SupportChain nextSupportLink;

	// public void __________(6)__________(SupportChain nextSupportLink) {
	public void setNext(SupportChain nextSupportLink) {
		this.nextSupportLink = nextSupportLink;
	}

	// public __________(7)__________ void __________(8)__________(IssueType issueType);
	public abstract void handleIssue(IssueType issueType);
}

File: DesignPatternsExercises/ruby/chain_of_responsibility_design_pattern/solution/support_chain.rb
# class __________(4)__________
class SupportChain
  # def __________(5)__________(chain)
  def next(chain)
    # __________(6)__________ = chain
    @next_support_link = chain
  end

  def handle_issue(issue_type)
    raise NotImplementedError, 'Subclasses must override this method'
  end
end
